### ISSUE TITLE
Invalid printf placeholders in internetgame and crash

### ISSUE THREAD

**Issue by frankystone** [widelands/widelands#5584]
After configuring a multiplayer game this morning the game crashed. The console output said:

~~~~
[00:00:01.078 real] INFO: **** END SOUND REPORT ****
[00:00:07.400 real] ERROR: [BufferedConnection] Trying to connect to 2a03:4000:32:524:48cb:feff:feae:b0c6:7395 failed!
[00:00:07.400 real] ERROR: [NetClient]: Trying to connect to 2a03:4000:32:524:48cb:feff:feae:b0c6:7395 failed!
[01:43:32.266 real] ERROR: format error: A string contains invalid printf placeholders
[01:43:32.266 real] ERROR: Error: [/home/kaputtnik/Quellcode/widelands-repo/widelands/src/base/format/tree.h:134] Too many arguments provided (expected only 0)
[01:43:32.266 real] ERROR: String: 1.1~git1 (beedd78@master)
[01:43:32.267 real] ERROR: format error: A string contains invalid printf placeholders
[01:43:32.267 real] ERROR: Error: [/home/kaputtnik/Quellcode/widelands-repo/widelands/src/base/format/tree.h:134] Too many arguments provided (expected only 0)
[01:43:32.267 real] ERROR: String: the-x1
[01:44:44.318 real] ERROR: format error: A string contains invalid printf placeholders
[01:44:44.318 real] ERROR: Error: [/home/kaputtnik/Quellcode/widelands-repo/widelands/src/base/format/tree.h:134] Too many arguments provided (expected only 0)
[01:44:44.318 real] ERROR: String: 1.1~bzr10379-202209071446
[01:44:44.318 real] ERROR: format error: A string contains invalid printf placeholders
[01:44:44.318 real] ERROR: Error: [/home/kaputtnik/Quellcode/widelands-repo/widelands/src/base/format/tree.h:134] Too many arguments provided (expected only 0)
[01:44:44.318 real] ERROR: String: WorldSavior
[01:44:44.517 real] ERROR: format error: A string contains invalid printf placeholders
[01:44:44.517 real] ERROR: Error: [/home/kaputtnik/Quellcode/widelands-repo/widelands/src/base/format/tree.h:134] Too many arguments provided (expected only 0)
[01:44:44.517 real] ERROR: String: maps/Downloaded/Inselkrieger Tournament version.wmf
[01:44:44.722 real] ERROR: format error: A string contains invalid printf placeholders
[01:44:44.722 real] ERROR: Error: [/home/kaputtnik/Quellcode/widelands-repo/widelands/src/base/format/tree.h:134] Too many arguments provided (expected only 0)
[01:44:44.722 real] ERROR: String: maps/Downloaded/Inselkrieger Tournament version.wmf
/usr/src/debug/gcc-build/x86_64-pc-linux-gnu/libstdc++-v3/include/bits/basic_string.h:1304: std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::back() [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; reference = char&]: Assertion '!empty()' failed.
~~~~

**To Reproduce**
Don't know. After starting widelands again except of those printf errors all worked fine.

**Version:**
 - OS: archlinux
 - Widelands Version: 1.1~git25864 (b301eaa@HEAD) Debug

**Additional context**

**Comment by tothxa** [comment]
I noticed the same in the previous crash logs for #5548 and #5555. Upon investigation I found the cause of these error messages, and prepared the fix as part of #5560. I'm not sure though whether the crashes are related to the misuse of try/catch, or is the PR merely an æsthetical cleanup.

**Comment by Noordfrees** [comment]
I don't think this crash is related to the format warnings. A backtrace for the assert would be useful, but this probably sounds like a rare race condition…
